dotCMS/src/main/java/com/dotmarketing/portlets/structure/action/EditFieldAction.java

```diff
@@ -114,71 +115,80 @@ public void processAction(ActionMapping mapping, ActionForm form, PortletConfig
         if ((cmd == null) || !cmd.equals("reorder")) {
             _retrieveField(form, req, res);
         }
-        HibernateUtil.startTransaction();
+
+        try {
+            HibernateUtil.startTransaction();
 
         /*
          * saving the field
          */
-        if ((cmd != null) && cmd.equals(Constants.ADD)) {
-            try {
-                Logger.debug(this, "Calling Add/Edit Method");
-
-                Field field = (Field) req.getAttribute(WebKeys.Field.FIELD);
-                if (InodeUtils.isSet(field.getInode())) {
-                    if (field.isFixed()) {
-                        
-                        FieldForm fieldForm = (FieldForm) form;
-                        field.setFieldName(fieldForm.getFieldName());
-                        field.setHint(fieldForm.getHint());
-                        field.setDefaultValue(fieldForm.getDefaultValue());
-                        field.setSearchable(fieldForm.isSearchable());
-                        field.setListed(fieldForm.isListed());
-
-                        com.dotcms.contenttype.model.field.Field newField = APILocator.getContentTypeFieldAPI().find(field.getIdentifier());
-                        if(LegacyFieldTypes.CONSTANT.implClass().getCanonicalName().equals(newField.typeName()) ||  
-                                        LegacyFieldTypes.HIDDEN.implClass().getCanonicalName().equals(newField.typeName())){
-                            field.setValues(fieldForm.getValues());
+            if ((cmd != null) && cmd.equals(Constants.ADD)) {
+                try {
+                    Logger.debug(this, "Calling Add/Edit Method");
+
+                    Field field = (Field) req.getAttribute(WebKeys.Field.FIELD);
+                    if (InodeUtils.isSet(field.getInode())) {
+                        if (field.isFixed()) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)